### ISSUE TITLE
Binaries compiled for glibc fail with "not found"

### ISSUE THREAD

**Issue by iamrecursion** [ish-app/ish#355]
When attempting to `make install` the GHC bindist, the install process fails saying that a file (`utils/ghc-cabal/dist-install/build/tmp/ghc-cabal`) is not found. The file definitely exists. It's quite a confusing one, as I don't see why it would happen. 

You can reproduce as follows, assuming a fresh installation of iSH:

1. `apk add coreutils findutils binutils bash gcc sed tar curl`
2. `curl -sSL https://downloads.haskell.org/~ghc/8.6.4/ghc-8.6.4-i386-deb9-linux.tar.xz > ghc-8.6.4.tar.xz`
3. `tar -xvf ghc-8.6.4.tar.xz`
4. `cd ghc-8.6.4`
5. `./configure` (this will fail if you have `gawk` installed with an error about not being able to open a file `-` for writing) 
6. `make install`

The only output in dmesg from a running `make install` alone is 51 (`dmesg | wc -l`) repetitions of `unknown ioctl 5413`. 

I know that this isn't a lot to go on so far, so I'm mainly opening this PR in order to potentially get some pointers for how to investigate further.

**Comment by tbodt** [comment]
This is the same as the issue with rust. I'll turn this into a tracking issue.

**Comment by iamrecursion** [comment]
Oh, that's a brilliant pointer. I'll try and get it working in a glibc chroot and report back (if I can get the chroot process to work with this update).

**Comment by tbodt** [comment]
The solution is to either find a way to build glibc for 32-bit musl, or switch to ubuntu. Neither is really easy.

**Comment by iamrecursion** [comment]
To update this issue with chroot progress, I followed [this](https://wiki.alpinelinux.org/wiki/Running_glibc_programs#Debian) guide to create a chroot, but it currently fails with a missing syscall 191. Issue has been filed separately for that.

**Comment by econcz** [comment]
This is a working solution: https://hub.docker.com/r/econcz/x86-alpine-glibc
I managed to install Miniconda3 and Stata 15 IC (32-bit) on this image in Docker, both are working.

**Comment by MasonProtter** [comment]
@econcz how did you get docker working?

**Comment by econcz** [comment]
Actually the apps are not working, they return illegal instructions (but it’s a problem inside the iSH App from what I understand since everything works in Docker).
Please check the iSH-project wiki on Github, there are two entries on how to install a compatible glibc-based filesystem in Docker and transfer it to the iSH App. Good luck!

**Comment by MasonProtter** [comment]
Okay, so the docker image was on running on an iOS device in iSH? I read the wiki pages, but they were not very illuminating.

**Comment by econcz** [comment]
> Okay, so the docker image was on running on an iOS device in iSH? I read the wiki pages, but they were not very illuminating.
> 
> My understanding from #63 was that docker was not yet functional on iSH

No-no, everything is much simpler: 1) you need to install Docker on your PC/Mac/Linux machine, 2) pull the image with glibc that I created from the Docker Hub, 3) create a container from it, 4) export the container as filesystem, 5) import the filesystem into iSH.

All the commands are described here:
[https://github.com/ish-app/ish/wiki/Transferring-filesystems-between-iSH-and-Docker](https://github.com/ish-app/ish/wiki/Transferring-filesystems-between-iSH-and-Docker)

**Comment by grassjelly** [comment]
> This is a working solution: https://hub.docker.com/r/econcz/x86-alpine-glibc
> I managed to install Miniconda3 and Stata 15 IC (32-bit) on this image in Docker, both are working.

Hey @econcz Thanks for sharing the docker file. I'm trying to install miniconda following this docker file https://github.com/Docker-Hub-frolvlad/docker-alpine-miniconda3/blob/master/Dockerfile but I'm getting an error `miniconda.sh: line 353: /home/conda/conda.exe: No such file or directory .
`

Just wondering if you happen to have a docker file available on how to install miniconda ? Thank you!

**Comment by econcz** [comment]
> > This is a working solution: https://hub.docker.com/r/econcz/x86-alpine-glibc
> > I managed to install Miniconda3 and Stata 15 IC (32-bit) on this image in Docker, both are working.
> 
> Hey @econcz Thanks for sharing the docker file. I'm trying to install miniconda following this instruction on the README https://github.com/ish-app/ish/wiki/Transferring-filesystems-between-iSH-and-Docker and this docker file https://github.com/Docker-Hub-frolvlad/docker-alpine-miniconda3/blob/master/Dockerfile but I'm getting an error `miniconda.sh: line 353: /home/conda/conda.exe: No such file or directory . `
> 
> Just wondering if you happen to have a docker file available on how to install miniconda ? Thank you!

Hi! You need to alter the miniconda.sh script somewhere in the beginning where it checks for the system type. It doesn't recognize Alpine as a Linux system and thinks you're on Windows (I remember running into this problem as well).
Personally, I wasn't able to run Miniconda 3 on iSH: the `conda` command produced illegal instructions (I guess the iSH's i686 emulation is not the same as a physical i686 so not all software is supported). But it will work in Docker.
There's a variety of iOS apps which support Jupyter with C-based modules (mostly, numpy, pandas, scikits etc.) that you can check out.

**Comment by grassjelly** [comment]
Thanks @econcz . I managed to install miniconda by using an older version (4.5.12) but encountered similar Illegal Instruction errors. Guess gotta have to wait until this issue gets fixed. Thank you!

**Comment by econcz** [comment]
I updated my post, please check the last sentence. Juno is a nice app.

**Comment by econcz** [comment]
Updated iSH glibc to 3.14
[https://hub.docker.com/r/econcz/x86-alpine-glibc/tags](https://hub.docker.com/r/econcz/x86-alpine-glibc/tags)
